code/_core/obj/hud/inventory/interaction.dm

```diff
@@ -94,8 +111,21 @@
 			drop_item_from_inventory(get_turf(src))
 		return TRUE
 
-	if(grabbed_object && grabbed_object == object)
-		release_object(caller)
+	if(grabbed_object && grabbed_object == object && is_living(grabbed_object)) //We click on the hand that grabbed something
+		if(world.time <= grab_time+DECISECONDS_TO_TICKS(20)) //Prevents insta agressive-grab
+			caller.to_chat(span("warning","Too soon to tighting your grip on [object]! Need to wait another [TICKS_TO_SECONDS(grab_time+DECISECONDS_TO_TICKS(20) - world.time)] seconds!"))
```

Comment:
A to_chat shouldn't be here. On 99% of features, we don't ever tell the player if they did something too soon.